CMakeLists.txt

```diff
@@ -50,6 +51,8 @@ if(NOT CUKE_DISABLE_BOOST_TEST)
     set(Boost_USE_STATIC_LIBS OFF)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DBOOST_TEST_DYN_LINK")
     find_package(Boost ${BOOST_MIN_VERSION} COMPONENTS unit_test_framework)
+    # Save the unit test library in Boost_LIBRARIES for use in tests/CMakeLists.txt
```

Comment:
Because the project's `CMakeLists.txt` calls `find_package(Boost ...)` multiple times, `Boost_LIBRARIES` can't be used to figure out where the Boost unit test framework is. Since cucumber-cpp always links this library dynamically, we need to copy it to the build dir when we are building a shared library. Otherwise, there are a bunch of runtime errors which occur on Windows because it won't be able to find that DLL when running the tests.